tools/dataproc-edge-node/util/setup_edge_node.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run as root on Dataproc master node after dataproc-startup-script
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
```

Comment:
In general it's good to avoid mixing variables inside strings, the preferred way is to use + and quote the variable. Something like echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: " + "${@}" >&2

Reply:
I separated the timestamp but you're going to need to provide a citation if you're suggesting to replace variables in quotes with +=